src/main/scala/firrtl/passes/Checks.scala

```diff
@@ -289,7 +256,10 @@ object CheckHighForm extends Pass with LazyLogging {
     }
     sinfo = c.info
     if (numTopM != 1) errors.append(new NoTopModuleException(c.main))
-    if (errors.nonEmpty) throw new PassExceptions(errors)
+    if (errors.nonEmpty) {
+      errors.append(new PassException(s"${errors.length} high form error(s) detected"))
+      throw new PassExceptions(errors)
```

Comment:
The error reporting idiom

    if (errors.nonEmpty)
      throw new PassExceptions(errors)

and perhaps

    if (errors.nonEmpty) {
      errors.append(...)
      throw new PassExceptions(errors)
    }

are probably common enough to justify utility functions (e.g., throwPassExceptions(errors), which does nothing if errors is empty).  Or perhaps make errors a class rather than ArrayBuffer[PassException]